agent/engine/docker_task_engine.go

```diff
@@ -474,7 +474,10 @@ func (engine *DockerTaskEngine) GetTaskByArn(arn string) (*api.Task, bool) {
 
 func (engine *DockerTaskEngine) pullContainer(task *api.Task, container *api.Container) DockerContainerMetadata {
 	pullStart := time.Now()
-	defer seelog.Infof("Finished pulling container %v. Lock acquisition and pull took %v.", container, time.Since(pullStart).String())
+	defer func(startTime time.Time) {
+		seelog.Infof("Finished pulling container %v. Lock acquisition and pull took %v.", container, time.Since(startTime).String())
```

Comment:
The duration here also included the time to wait for the lock, I think it is more clear to put there?